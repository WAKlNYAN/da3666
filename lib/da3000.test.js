const {
  addUser,
  addComment,
  store,
} = require('./da3000');

test('User can be created', () => {
  return addUser('Bob').then(data => {
    const {
      success,
      affectedEntity: {
        type,
        userName,
        sources,
        id,
        date: {
          created,
        },
      },
    } = data;

    expect(success).toBeTruthy();
    expect(type).toBe(0);
    expect(userName).toBe('Bob');
    expect(sources).toContain(0);
    expect(id).toBe(0);
    expect(created).toBeTruthy();
  });
});

test('Second user can be created', () => {
  return addUser('Sue').then(data => {
    const {
      success,
      affectedEntity: {
        type,
        userName,
        sources,
        id,
        date: {
          created,
        },
      },
    } = data;

    expect(success).toBeTruthy();
    expect(type).toBe(0);
    expect(userName).toBe('Sue');
    expect(sources).toContain(0);
    expect(id).toBe(1);
    expect(created).toBeTruthy();
  });
});

test('Bob can comment', () => {
  const testText = 'Hello, sue.';
  const testParentId = 1; // Bob is commenting on Sue's user entity
  return addComment({
    text: testText,
    parentId: testParentId
  }).then(({
    success,
    affectedEntity: {
      type,
      text,
      sources,
      id,
      parentId,
      date: {
        created,
      },
    },
  }) => {
    expect(success).toBeTruthy();
    expect(type).toBe(0);
    expect(text).toBe(testText);
    expect(sources).toContain(0);
    expect(id).toBe(2);
    expect(parentId).toBe(testParentId);
    expect(created).toBeTruthy();
  });
});

test('Entity snapshot is correct', () => {
  store.entities.forEach((entity) => {
    expect(entity).toMatchSnapshot(
      {
        date: {
          created: expect.any(Date),
        }
      }
    );
  });
});
