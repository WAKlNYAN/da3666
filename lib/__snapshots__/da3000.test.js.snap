// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Entity snapshot is correct 1`] = `
Object {
  "date": Object {
    "created": Any<Date>,
  },
  "id": 0,
  "sources": Array [
    0,
  ],
  "type": 0,
  "userName": "Bob",
}
`;

exports[`Entity snapshot is correct 2`] = `
Object {
  "date": Object {
    "created": Any<Date>,
  },
  "id": 1,
  "sources": Array [
    0,
  ],
  "type": 0,
  "userName": "Sue",
}
`;

exports[`Entity snapshot is correct 3`] = `
Object {
  "date": Object {
    "created": Any<Date>,
  },
  "id": 2,
  "parentId": 1,
  "sources": Array [
    0,
  ],
  "text": "Hello, sue.",
  "type": 0,
}
`;
